face/src/plugins/timeline/TimelineView.js

```diff
@@ -5,24 +5,44 @@ import {className} from '../../decorators';
 import duration from '../../helpers/duration';
 
 import {scaleLinear, scaleBand} from 'd3-scale';
+import {event as currentEvent} from 'd3-selection';
+import {brushX} from 'd3-brush';
 import 'd3-selection-multi';
 
 import escape from '../../util/escape';
 import TooltipView from '../../components/tooltip/TooltipView';
 
-const BAR_HEIGHT = 30;
-const BAR_GAP = 1;
+const BAR_HEIGHT = 25;
+const BAR_GAP = 0.5;
+const MINIBAR_HEIGHT=5;
 
 const PAD_BOTTOM = 30;
-const PAD_LEFT = 10;
-const PAD_RIGHT = 10;
+const PAD_LEFT = 25;
+const PAD_RIGHT = 25;
+
+const MINI_HEIGHT = 150;
 
 @className('timeline')
 class TimelineView extends BaseChartView {
     initialize() {
         this.x = scaleLinear();
+        this.miniX = scaleLinear();
         this.tooltip = new TooltipView({position: 'bottom'});
-    }
+
+        this.brush = brushX()
+            .on('brush', () => {
+                var selection = currentEvent.selection;
+                if (selection) { 
+                    var maxRight = this.x.range()[1];
+                    this.x.domain(selection.map(this.miniX.invert, this.miniX));
+                    this.svg.selectAll('.timeline__item').attrs({
+                        x: d => Math.max(0, Math.min(this.x(d.start, maxRight))),
```

Comment:
`Math.min()` shouldn't be called with one argument.
Possibly should be
```
Math.min(this.x(d.start), maxRight)
```

See parenthesis position